lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -953,9 +968,13 @@ public void checkCanExecuteFunction(SystemSecurityContext systemSecurityContext,
     @Override
     public void checkCanExecuteFunction(SystemSecurityContext systemSecurityContext, FunctionKind functionKind, CatalogSchemaRoutineName functionName)
     {
-        if (functionKind == TABLE) {
-            denyExecuteFunction(functionName.toString());
+        switch (functionKind) {
+            case SCALAR, AGGREGATE, WINDOW:
+                return;
+            case TABLE:
+                denyExecuteFunction(functionName.toString());
```

Comment:
Add 
>  // TODO implement

?

Reply:
`Make checkCanExecuteFunction checks future proof` is meant to be just a code refactor.